data-prepper-plugins/common/README.md

```diff
@@ -8,9 +8,33 @@ A prepper plugin to generate new string records with upper or lower case convers
 
 ## `file` (source)
 
-A source plugin to read input data from the specified file path.
-
-- path (String): absolute input data file path
+A source plugin to read input data from the specified file path. The file source creates a new Record for each line of data in the file.
+
+* `path` (String): absolute input data file path. It is required
+
+* `format` (String): The format of each line of the file. Valid options are `json` or `plain`. Default is `plain`.
+  <br></br>
+    * `plain`: Reads plaintext data from files. Internally, a plain text line from a file will be given a key of `message` as shown below.
+    ```
+    Example log line in file
+    ```
+  becomes 
+    ```
+    { "message": "Example log line in file" }
+    ```
+  
+    * `json`: Reads data that is in the form of a JSON string from a file. If the json string is unable to be parsed, the file source will treat it as a plaintext line.
+  Expects json lines as follows:
+      ```
+      { "key1": "val1" }
+      { "key2": "val2" }
+      { "key3": "val3" }
+      ```
+            
+  
+* `record_type` (String): The Event type that will be stored in the metadata of the Event. Default is `event`. 
```

Comment:
Thanks for adding this, even though it is temporary.

I'd generally prefer to keep the default as `string` since that is the default behavior. But, since this is more for test and non-Trace Analytics pipelines, it could be a reasonable breaking change. If the default is `event`, the release notes should state this, and how to get past it. #618 .

Reply:
That's a good point. Probably smarter to keep the default as string.

Reply:
I will list it as a breaking change.